src/errors.ts

```diff
@@ -1,17 +1,19 @@
-'use strict'
-
-exports.messages = {
+export const messages = {
   NOT_STARTED_YET: 'The libp2p node is not started yet',
   DHT_DISABLED: 'DHT is not available',
   CONN_ENCRYPTION_REQUIRED: 'At least one connection encryption module is required',
+  ERR_TRANSPORTS_REQUIRED: 'At least one transport module is required',
+  ERR_PROTECTOR_REQUIRED: 'Private network is enforced, but no protector was provided',
   NOT_FOUND: 'Not found'
 }
 
-exports.codes = {
+export const codes = {
```

Comment:
enum?

Reply:
Enums end up as numbers at runtime, right?  the use case is:

```js
throw errCode(new Error(messages.FOO), codes.FOO)

// later
if (err.code === codes.FOO) {
  ...
}
```

If someone decided to re-order the enum (alphabetical, add a new code, whatever) this could cause very undesirable side effects in code that did (rightly or wrongly):

```js
if (err.code === 'FOO') {
  ...
}
```

Reply:
Non-ts code, I mean.  Maybe I'm worrying about nothing.

Reply:
Hmm we can use string enums so they don't end up being numbers and have same functionality as we have now: https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

And they shouldn't be replaced by value unless you add const: https://www.typescriptlang.org/docs/handbook/enums.html#const-enums

Reply:
Ah, nice - I've updated it.